crates/sc-consensus-subspace/src/archiver.rs

```diff
@@ -610,18 +610,33 @@ where
     // If there is no path to this block from the tip due to snap sync, we'll start archiving from
     // an earlier segment, then start mapping again once archiving reaches this block.
     if let Some(block_number) = create_object_mappings.block() {
-        best_block_to_archive = best_block_to_archive.min(block_number);
+        // There aren't any mappings in the genesis block, so starting there is pointless.
+        // (And causes errors on restart, because genesis block data is never stored during snap sync.)
+        best_block_to_archive = best_block_to_archive.min(block_number).max(1);
```

Comment:
Block is already constrained to be non-zero in `create_object_mappings`, so not sure how much this is actually needed